### PR TITLE
Feature classifier guide

### PR DESCRIPTION
**Pull request by Yongyao**
https://github.com/ArcGIS/geosaurus/issues/2390

Added credit to fast.ai

### CODE REVIEW

**Comment by CMPeng** [comment]
Good job with the guides @Yongyao 
please find the revision suggestions down below (again those are just suggestions. If you have your own design, please ignore...):

1. For the feature_categorization.ipynb, suggestions are the same as the those in the sample PR.

2. For the how-feature-classifier-works.ipynb, there are some grammar related issues, e.g. 
 - [ ] please rewrite "A feature class that defines the location (e.g. outline or bounding box) each feature"
 - [ ] please rewrite "Once the training samples are exported, it can be used the training input for the deep learning based classification algorithm to train a feature classifier."
 - [ ] please rewrite "Optionally, we can also define a buffer size to extract a larger surrounding area that the feature extent so more spatial context is available to the classification model, which makes distinguishing different classes easier."
 - [ ] please rewrite "In the example above, there are there buildings in the original data. The top one is damaged and the other two is undamanged"
 - [ ] please rewrite "Most of them is a backbone CNN architecture (e.g. Resnet, LeNet-5, AlexNet, VGG 16) followed by a softmax layer."
 - [ ] "data is the returned data object from prepare_data function. backbone is used for creating the base of the classifier, which is resnet34 by default, while pretrained_path points to where pre-trained model is saved." can be changed something like,
"in which data is the data object returned from the prepare_data function, backbone is ... (which is ResNet34 by default) while pretrained_path ... is saved."

3. In how-unet-works.ipynb,
 - [ ] "You can tweak everything in the encoder and our U-net module creates decoder equivalent to that [2] ." you might want to correspond back to the reason of this discussion, e.g.
"since that you can ..., the initiation of Unetclassifier requires fewer parameters..."

**Comment by AtmaMani** [comment]
<p>@yongyao please find my suggestions below (these are the same I posted on the PR you sent to DSX repo)</p><p> </p><p>feature_categorization notebook</p><ol><li>Introduction talks about publishing the model, but in this guide we are not. So intro needs to be updated.</li><li>somehow the url of the tile layer in IVT portal that Romero published has an extra 'home' in the URL. Can you rerun this cell so the url is updated?</li><li>Export training data section: enclose code terms such as tile_size, chip_format etc within code ticks <code></code></li><li>typo in strid_size</li><li>Change to "Depending on the size of your data, tile and stride size, and computing resources, this operation can take a while. In our experiments, this took 15mins~2hrs"</li><li>Train a model.. section: add plural to word rates in "Here we explore a range of learning <strong>rate</strong> to guide us to choose the best one."</li><li>Inference part: why not call the export training data as you did in the training part?</li></ol><p> </p>

**Comment by Yongyao** [comment]
@AtmaMani I addressed all comments except "Inference part: why not call the export training data as you did in the training part?". Not sure what you mean.

**Comment by AtmaMani** [comment]
@Yongyao can you showcase using `categorize_features()` instead of `classify_features()` (following the email thread on the same)?

**Comment by Yongyao** [comment]
@AtmaMani These two methods have different signatures/parameters. With `classify_features()`, there are a few tricks we need to do. The other one seems easier, but I'm not sure if it is supported yet. Checking with Rohit through email.

**Comment by Yongyao** [comment]
@AtmaMani I've changed it to `categorize_features`. Turns out there is a bug. Sandeep and I have found a workaround by loading the model using emd file explicitly. He is going to investigate further and submit an enhancement.
